### PR TITLE
✨ add auth0 management actions

### PR DESCRIPTION
**Pull request by jgnieuwhof**
## Motivation
Auth0 recently disabled wildcards for 'Allowed Web Origins', meaning that we can no longer whitelist all netlify preview builds (i.e. https://*--zeus-api.netlify.com). Each preview build needs to be added to the list of allowed origins, and removed once it's no longer in use.

## Approach
I've added two actions: `auth0-add-origin`, and `auth0-remove-origin`. One adds a URL to an auth0 client, the other removes it, respectively. Documentation on the actions is provided in their README files.

Since these actions are javascript I've turned this repository into a yarn workspace. Since actions are resolved from the project structure they've remained top level entities. Internal packages used by the actions can be found in `core`.

Again, since actions are resolved from the GitHub project structure, we need to check in their build. To avoid checking in the entire `node_modules` tree each Javascript action can use [`@zeit/ncc`](https://github.com/zeit/ncc) to build its entire dependency tree into a single distributable `index.js` file.

## Example Usage
https://github.com/resideo/zeus/pull/429

## Questions
It's currently on the developer to ensure that `yarn build` is run prior to check-in, should we use some sort of hook to do this?

### CODE REVIEW

File: auth0-add-origin/dist/index.d.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export {};
```
Comment (taras) [inline auth0-add-origin/dist/index.d.ts]: Is this file supposed to be included?

File: auth0-add-origin/dist/index.d.ts.map [hunk 2]
```diff
@@ -0,0 +1 @@
+{"version":3,"file":"","sourceRoot":"","sources":["file:///Users/jgnieuwhof/projects/resideo/repos/actions/auth0-add-origin/index.ts"],"names":[],"mappings":""}
```
Comment (taras) [inline auth0-add-origin/dist/index.d.ts.map]: Is this supposed to be included?

File: auth0-add-origin/dist/index.d.ts (hunk 1 above)
Comment (jgnieuwhof) [reply auth0-add-origin/dist/index.d.ts]: It's not necessary, but `ncc` provides no option to exclude them from the build. I can exclude them from `git` using `.gitignore` if you think that'd be best?

File: auth0-add-origin/dist/index.d.ts.map (hunk 2 above)
Comment (jgnieuwhof) [reply auth0-add-origin/dist/index.d.ts.map]: See comment above.

**Comment by tazsingh** [comment]
This repo was a bit off-the-radar for me as it's not in ZenHub - I found it when reading the PR description on https://github.com/resideo/admin-portal/pull/19 as I believe merging this in is a blocker to that? What are your thoughts on adding this repo to ZenHub so it can follow the same process?

**Comment by tazsingh** [comment]
I'm not overly enthused about checking in built files, but I do understand your reasoning. I'm wondering if @shanewilson is aware of perhaps a different way of doing this that wouldn't require checking in the files? Or maybe not using TypeScript as I'm assuming that's what's requiring the build step?

**Comment by jgnieuwhof** [comment]
@tazsingh - good call, added this repo to the ZenHub workspace.

As per checking in dist - it is GitHub's recommended way of how to structure Javascript actions: https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-a-javascript-action#commit-tag-and-push-your-action-to-github.

I don't think we want to write all these in pure vanilla JS, for a couple reasons:
- We'd still 
. We lose a lot by not going with Typescript, and it goes against the practices that the team has set.

I'd much prefer some checked-in `dist` files

**Pull request merged**